here-naksha-lib-ext-manager/src/main/java/com/here/naksha/lib/extmanager/ExtensionCache.java

```diff
@@ -112,41 +112,70 @@ private void publishIntoCache(KVPair<Extension, File> result, ExtensionConfig ex
       if (!isNullOrEmpty(extension.getInitClassName())) {
         try {
           Class<?> clz = loader.loadClass(extension.getInitClassName());
-          clz.getConstructor(INaksha.class, Extension.class).newInstance(naksha, extension);
-        } catch (ClassNotFoundException
-            | InvocationTargetException
-            | InstantiationException
-            | NoSuchMethodException
-            | IllegalAccessException e) {
+          Object obj = clz.getConstructor(INaksha.class, Extension.class).newInstance(naksha, extension);
+          if (obj instanceof IExtensionInit initInstance) {
+            initInstance.close();
```

Comment:
why close() first? we would only call init()

Reply:
It was kept for the safer side to make sure every instance is closed before doing next init, if we dont have any instance running it will just log an error message. But yes we can remove it.

Reply:
Done.